OISF/suricata
# Pcre2 3194 v13

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3194

Describe changes:
- migrate all pcre1 code to pcre2
- fuzz: builds fuzz_sigpcap_aware only when asked (ie ./configure --enable-fuzztargets)
- qa : remove some obsolete files
- pcrexform : checks that we have exactly one substring capture in a signature
- transforms : use generic `Free` functions everywhere
suricata-verify-pr: 480

https://github.com/OISF/suricata-verify/pull/480

Replaces https://github.com/OISF/suricata/pull/6052
- commits reordered
- fix make check with handling `PCRE2_ERROR_UNSET

This is ready for review (and it needs a big review and testing)
Here are the big questions beyond details :
- Should I make a PR with the first 4 preliminary commits ? (like transform: use generic Free function)
- Should we include in this PR the removal of the usage of PCRE for some keywords parsing like https://redmine.openinfosecfoundation.org/issues/4112 #5966 
- There are some changes in the behavior between PCRE1 and PCRE2 (in the patterns `\i`, `[\w-.]`, or one the behavior to get substrings (now returning an error instead of an empty string if the capture group is not there). How would we deal with this ?
